src/condition/ssrf_detector.cpp

```diff
@@ -150,18 +140,43 @@ bool detect_parameter_injection(
     return false;
 }
 
+bool is_forbidden_uri(const uri_decomposed &uri,
+    const std::unique_ptr<matcher::ip_match> &forbidden_ip_matcher,
+    const std::unordered_set<std::string_view> &allowed_scheme_set,
+    const std::vector<std::string> &forbidden_domains)
+{
+    const auto is_forbidden_domain = [&](std::string_view host) {
+        for (const auto &domain : forbidden_domains) {
+            if (host.ends_with(domain)) {
+                return true;
+            }
+        }
+        return false;
+    };
+
+    const auto is_forbidden_ip = [&](const std::optional<ipaddr> &ip) {
+        return ip.has_value() && forbidden_ip_matcher->match_ip(ip.value());
+    };
+
+    return !allowed_scheme_set.contains(uri.scheme) || is_forbidden_domain(uri.authority.host) ||
+           is_forbidden_ip(uri.authority.host_ip);
+}
+
 ssrf_result ssrf_impl(const uri_decomposed &uri, const ddwaf_object &params,
     const exclusion::object_set_ref &objects_excluded, const object_limits &limits,
-    const std::unique_ptr<matcher::ip_match> &dangerous_ip_matcher,
-    const std::unordered_set<std::string_view> &authorised_scheme_set, ddwaf::timer &deadline)
+    const ssrf_opts &opts, const std::unique_ptr<matcher::ip_match> &forbidden_ip_matcher,
+    const std::unordered_set<std::string_view> &allowed_scheme_set,
+    const std::vector<std::string> &forbidden_domains, ddwaf::timer &deadline)
 {
     static constexpr std::size_t min_str_len = 4;
 
-    std::string_view dangerous_domain = {};
-    for (const auto domain : dangerous_domains) {
-        if (uri.authority.host.ends_with(domain)) {
-            dangerous_domain = domain;
-            break;
+    std::string_view forbidden_domain = {};
+    if (!opts.enforce_policy_without_injection) {
```

Comment:
if `enforce_policy_without_injection` is true, this method shouldn't even be reached, right?